app/repositories/SessionRepository.scala

```diff
@@ -16,66 +16,54 @@
 import models.UserAnswers
+import org.mongodb.scala.model.{Filters, IndexModel, IndexOptions, ReplaceOptions}
 import play.api.Configuration
 import play.api.libs.json._
-import play.modules.reactivemongo.ReactiveMongoApi
-import reactivemongo.api.indexes.IndexType
-import reactivemongo.play.json.collection.Helpers.idWrites
-import reactivemongo.play.json.collection.JSONCollection
+import uk.gov.hmrc.mongo.MongoComponent
+import uk.gov.hmrc.mongo.play.json.PlayMongoRepository
 
+import java.time.LocalDateTime
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
 
-class DefaultSessionRepository @Inject()(
-                                          val mongo: ReactiveMongoApi,
-                                          val config: Configuration
+class DefaultSessionRepository @Inject()(val mongo: MongoComponent,
+                                         val config: Configuration
                                         )(implicit val ec: ExecutionContext)
-  extends SessionRepository
-    with IndexManager {
-
-  override val collectionName: String = "user-answers"
-
-  private val cacheTtl = config.get[Int]("mongodb.timeToLiveInSeconds")
-
-  private def collection: Future[JSONCollection] = for {
-    _ <- ensureIndexes
-    col <- mongo.database.map(_.collection[JSONCollection](collectionName))
-  } yield col
-
-  private val lastUpdatedIndex = MongoIndex(
-    key = Seq("lastUpdated" -> IndexType.Ascending),
-    name = "user-answers-last-updated-index",
-    expireAfterSeconds = Some(cacheTtl)
-  )
-
-  private def ensureIndexes: Future[Unit] = for {
-    col <- mongo.database.map(_.collection[JSONCollection](collectionName))
-    _ <- col.indexesManager.ensure(lastUpdatedIndex)
-  } yield ()
-
-  override def get(id: String): Future[Option[UserAnswers]] =
-    collection.flatMap(_.find(Json.obj("_id" -> id), None).one[UserAnswers])
+  extends PlayMongoRepository[UserAnswers](
+    mongoComponent = mongo,
+    collectionName = "user-answers",
+    domainFormat = Format(UserAnswers.reads, UserAnswers.writes),
+    indexes = Seq(
+      IndexModel(
+        Indexes.ascending("lastUpdated"),
+        IndexOptions().name("user-answers-last-updated-index")
+          .expireAfter(config.get[Int]("mongodb.timeToLiveInSeconds"), TimeUnit.SECONDS)
+          .unique(false))
+    ),
+    replaceIndexes = config.getOptional[Boolean]("microservice.services.features.mongo.dropIndexes").getOrElse(false)
+  ) with SessionRepository {
+
+  private def byId(id: String) = Filters.eq("_id", id)
+
+  override def get(id: String): Future[Option[UserAnswers]] = {
+    val modifier = Updates.set("lastUpdated", LocalDateTime.now)
+
+    val options = new FindOneAndUpdateOptions()
+      .upsert(false)
+      .returnDocument(ReturnDocument.AFTER)
+    collection.findOneAndUpdate(byId(id), modifier, options).headOption()
```

Comment:
Is there a reason for updating the "lastUpdated" value in this method? Previously it was a read only

Reply:
If user start to answer questions and dont finish in 15 mins timeframe then they wont be able to submit even if they submited one question per 3 mins for example just as slow pace, in this case we will update usernaswers everytime system request it and then user will have 15 mins timeout only when they are really away for 15 mins, hope that makes sense the same logic is in other services

Reply:
Cool yeah that makes sense, ready to merge then!